pkg/kubectl/cmd/rollout/rollout_pause.go

```diff
@@ -89,32 +90,44 @@ func (o *PauseConfig) CompletePause(f *cmdutil.Factory, cmd *cobra.Command, out
 		return err
 	}
 
-	infos, err := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+	r := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
 		ResourceTypeOrNameArgs(true, args...).
-		SingleResourceType().
+		ContinueOnError().
 		Latest().
-		Do().Infos()
+		Flatten().
+		Do()
+	err = r.Err()
 	if err != nil {
 		return err
 	}
-	if len(infos) != 1 {
-		return fmt.Errorf("rollout pause is only supported on individual resources - %d resources were found", len(infos))
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		o.Infos = append(o.Infos, info)
+		return nil
+	})
+	if err != nil {
+		return err
 	}
-	o.Info = infos[0]
 	return nil
 }
 
 func (o PauseConfig) RunPause() error {
-	isAlreadyPaused, err := o.PauseObject(o.Info.Object)
-	if err != nil {
-		return err
-	}
-	if isAlreadyPaused {
-		cmdutil.PrintSuccess(o.Mapper, false, o.Out, o.Info.Mapping.Resource, o.Info.Name, "already paused")
-		return nil
+	allErrs := []error{}
+	for _, info := range o.Infos {
+		isAlreadyPaused, err := o.PauseObject(info.Object)
+		if err != nil {
+			allErrs = append(allErrs, err)
```

Comment:
In other commands, you see `cmdutil.AddSourceToError` called so that the messages can show which resources they failed on.